arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/transients/DeploymentInfo.java

```diff
@@ -32,28 +36,45 @@ public class DeploymentInfo {
    * @param address the address to get information for
    * @return the deployment number for this address
    */
-  public final int number(Address address) {
-    return this.deploymentNumber.getOrDefault(address, 0);
+  public final int deploymentNumber(Address address) {
+    return this.getDeploymentNumber(address);
+  }
+
+  public void newDeploymentAt(Address address) {
+    this.incrementDeploymentNumber(address);
+    this.markAsUnderDeployment(address);
   }
 
-  public void deploy(Address address) {
-    this.deploymentNumber.put(address, this.number(address) + 1);
-    this.markDeploying(address);
+  public void newDeploymentAtForTxSkip(Address address) {
+    this.incrementDeploymentNumber(address);
+    this.markAsNotUnderDeployment(address);
+  }
```

Comment:
`newDeploymentAtForTxSkip` is special as one immediately transitions to a deployed account with empty code.